lib/marked.js

```diff
@@ -16,7 +16,8 @@ var block = {
   code: /^( {4}[^\n]+\n*)+/,
   fences: noop,
   hr: /^ {0,3}((?:- *){3,}|(?:_ *){3,}|(?:\* *){3,})(?:\n+|$)/,
-  heading: /^ *(#{1,6}) *([^\n]+?) *(?:#+ *)?(?:\n+|$)/,
+  // cap[2] might be ' HEADING # ' and must be trimmed appropriately.
+  heading: /^ *(#{1,6})([^\n]*)(?:\n+|$)/,
```

Comment:
could we replace `[^\n]` with `.`?

Reply:
Yes haha

Reply:
Worth noting that this regex is substituted into `block.paragraph` and the follow-up parsing will not be performed there. I think we're OK spec-wise but might want to double-check.